### DISCUSSION TITLE
sass-loader Source Maps not accurate, shows bundled css

### DISCUSSION THREAD

**Discussion by da-kicks-87** [webpack/webpack#14098] [Q&A] [answered]
Hello,

I am trying to get the sass-loader Source Maps to work.
When using inspector on elements in my browser I do see the .scss files , but I also see the final bundled .css file. I don't think i should be seeing that. It doesn't seem accurate to me. 

Has anyone had this problem?

Here is my webpack config file:

```
const path = require('path');
const MiniCssExtractPlugin = require('mini-css-extract-plugin');
const CssMinimizerPlugin = require('css-minimizer-webpack-plugin');

module.exports = {
  mode: 'development',
  watch: true,
  devtool: 'source-map',
  entry: './src/js/index.js',
  output: {
    path: path.resolve(__dirname, 'dist/js'),
    filename: 'bundle.js'
  },
  optimization: {
    minimize: true,
    minimizer: [
    new CssMinimizerPlugin(),
    ],
  },
  plugins: [
      new MiniCssExtractPlugin({
      // Options similar to the same options in webpackOptions.output
      // both options are optional
      filename: '../css/master-compiled.css',
      chunkFilename: '[id].css',
    })],
  module: {
     rules: [
       {
        test: /\.s[ac]ss$/i,
        use: [
          {
            loader: MiniCssExtractPlugin.loader,
            options: {
              // you can specify a publicPath here
              // by default it uses publicPath in webpackOptions.output
              publicPath: '../',
            }
          },
        {
          loader: 'css-loader', // translates CSS into CommonJS modules
          options: {
              sourceMap: true,
            }
        },
        {
          loader: 'postcss-loader', // Run postcss actions
          options: {
            postcssOptions: {
              plugins: [ require('autoprefixer') ]
            }
          }
        },
        {
          loader: 'sass-loader', // compiles Sass to CSS
          options: {
              sourceMap: true,
            },
        }]
       },
     ],
   },

};
```

**Comment by alexander-akait** [comment, marked as answer]
Related https://github.com/webpack-contrib/mini-css-extract-plugin/issues/814?

**Reply by da-kicks-87** [reply]
I'm not sure if it's related. Here a a screenshot of what my issue is:

![sass-source-map-error](https://user-images.githubusercontent.com/70115804/131515071-1ac7b22c-53ea-40e4-ab60-cf0ad8ec0be6.JPG)

See how the **master-complied.cs**s is showing along with the **_grids.scss** ? The col-12 class should be shown coming from Bootstrap 5 .scss files.

**Reply by alexander-akait** [reply]
First remove `publicPath: '../',`, it is wrong and invalid, webpack v5 supports `publicPath: 'auto'` and it is default value, you don't need this dirty hack, also you don't need `sourceMap: true`, because you set `devtool: 'source-map',`

**Reply by da-kicks-87** [reply]
I cleaned up the config. Same issue appears.

**Reply by alexander-akait** [reply]
Can you provide reproducible test repo (link on github)? We have tests on source maps, so I think something wrong in setup

**Reply by da-kicks-87** [reply]
Here is a link to my repo, I made it public: 
[https://github.com/da-kicks-87/static-business-template](url)

**Reply by alexander-akait** [reply]
How I can run it?

**Reply by alexander-akait** [reply]
Fixed in upstreams, update all your deps to the latest stable versions, i.e. `webpack`/`sass`/`mini-css-extract-plugin`/`autoprefixer`, with updated deps the problems doesn't exist

**Reply by da-kicks-87** [reply]
Thank you! It works now after I updated the dependencies. I'm pretty sure I had updated all of them a few weeks ago.
